WcaOnRails/app/views/competitions/my_competitions.html.erb

```diff
@@ -8,29 +8,100 @@
+          <td><%= competition.cityName %>, <%= competition.country_name %></td>
+          <td><%= competition.start_date %></td>
+          <td>
+            <% if current_user.can_manage_competition?(competition) %>
+              <%= link_to "Edit", edit_competition_path(competition) %>
+            <% end %>
+          </td>
+          <td>
+            <% if competition.use_wca_registration? && current_user.can_manage_competition?(competition) %>
+              <%= link_to "Registrations", competition_edit_registrations_path(competition) %>
+            <% end %>
+          </td>
+        </tr>
       <% end %>
-      <% table.column do |competition| %>
-        <% if competition.use_wca_registration? && current_user.can_manage_competition?(competition) %>
-          <%= link_to "Registrations", competition_edit_registrations_path(competition) %>
+    </table>
+
+    <button id="show-hidden" class="btn btn-primary" type="button" data-toggle="collapse" data-target="#past-competitions" aria-expanded="false" aria-controls="collapseExample">Show past competitions</button>
+  <% end %>
+
+  <div id="past-competitions" class="collapse">
+    <h2>Past competitions</h2>
+    <% if @past_competitions.length < 1 %>
+      <div class="alert alert-info" role="alert">
+        You have no past competitions.
+      </div>
+    <% else %>
+      <table class="table competitions-table">
+        <tr>
+          <th>Name</th>
+          <th>Location</th>
+          <th>Start date</th>
+          <th></th>
+          <th></th>
+        </tr>
+        <% @past_competitions.each do |competition| %>
+          <tr>
+            <td><%= link_to competition.name, competition_path(competition) %></td>
+            <td><%= competition.cityName %>, <%= competition.country_name %></td>
+            <td><%= competition.start_date %></td>
+            <td>
+            <% if current_user.can_manage_competition?(competition) %>
+              <%= link_to "Edit", edit_competition_path(competition) %>
+            <% end %>
+            </td>
+            <td>
+              <% if competition.use_wca_registration? && current_user.can_manage_competition?(competition) %>
+            <%= link_to "Registrations", competition_edit_registrations_path(competition) %>
+          <% end %>
+            </td>
+          </tr>
         <% end %>
-      <% end %>
+      </table>
     <% end %>
-  <% end %>
+  </div>
 
-  <% if current_user.wca_id %>
-    <%= link_to "My profile", "/results/p.php?i=" + current_user.wca_id %>
-  <% end %>
+  <p>
+    <% if current_user.wca_id %>
+      <%= link_to "My profile", "/results/p.php?i=" + current_user.wca_id %>
+    <% end %>
+  </p>
 
 </div>
+
+<script>
+  (function() {
+    $("#show-hidden").click(function(e){
+      $(this).text(function(i, text){
+        return text === "Show past competitions" ? "Hide past competitions" : "Show past competitions";
+      })
+      $(this).toggleClass("btn-warning");
```

Comment:
I feel like the "right way" to do this is to add a css rule for the button (I believe it gets the collapsed state when the thing it controls is collapsed), rather than toggling this class ourselves via javascript.